### PR TITLE
Remove streams from cm on cm-janus shutdown

### PR DESCRIPTION
**Pull request by fvovan**

### CODE REVIEW

**Comment by fvovan** [comment]
@vogdb  could you review it please? 
not sure where to place streams cleanup code. `index.js` seems to be the most appropriate one.

**Comment by tomaszdurka** [comment]
By adding method to remove streams on index.js we are adding kinda global code - stream-specific code into entry point.

I am not sure if streams are the only ones we should remove. Maybe some plugins have other things which should be removed.
What about adding `stop` method into `janus-proxy` which would `remove` all conenctions? 
- connections would remove sessions
- sessions would remove plugins
- plugins would remove streams
We would be sure we cleaned up everything as we usually do, no need to add another usage of cm-http-client.

In general I think index.js should be global *entry point* which starts sub-apps (jobhandler, proxy, api-server) and stops all of them on incoming signal. I don't think it should know anything about streams in the end... except it needs to register service like that :/

**Comment by fvovan** [comment]
>In general I think index.js should be global entry point which starts sub-apps (jobhandler, proxy, api-server) and stops all of them on incoming signal. I don't think it should know anything about streams in the end... except it needs to register service like that :/

yes, i completely agree. just didn't know where to put it.
now it's not even needed, streams will be removed within plugins' deletion.
@tomaszdurka please take a look

File: lib/index.js [hunk 1]
```diff
@@ -62,7 +63,16 @@ Application.prototype.start = function() {
 
   process.on('uncaughtException', function(err) {
     serviceLocator.get('logger').error('Unexpected runtime error. Shutdown the process ' + err.stack || err.message);
-    jobManager.stop().finally(function() {
+    process.emit('close');
+  });
+
+  process.on('SIGINT', function() {
```
Comment (tomaszdurka) [inline lib/index.js]: Listen for SIGTERM too.

**Comment by tomaszdurka** [comment]
As in in-code comment - should we listen for SIGTERM?

Rest lgtm.

**Pull request merged**